src/java.xml/share/classes/javax/xml/xpath/XPathFactory.java

```diff
@@ -369,4 +369,56 @@ public abstract boolean getFeature(String name)
     * @return New instance of an <code>XPath</code>.
     */
     public abstract XPath newXPath();
+
+    /**
+     * Sets a property for this {@code XPathFactory} and {@code XPath}
+     * created by this factory.
```

Comment:
Are properties inherited by the XPath object at create time? The javadoc for newXPath suggests so and I'm wondering if the javadoc for setProperty needs to consistent with that. Maybe the intention is really that this method sets a XPathFactory property and it is used when configuring the XPath object returned by the newXPath method.

Reply:
Yes. Made a clarification to the javadoc, and while we are here, to setFeature method as well.

Reply:
One other suggestion for wording is to change:
"A property set on the factory is effective to the XPath object created thereafter"
to something like:
"The property applies to XPath objects that the XPathFactory creates. It has no impact on XPath objects that are already created."

I don't have any other comments on the API.

Reply:
The wording is adjusted accordingly. Thanks.

Reply:
> The wording is adjusted accordingly. Thanks.

Thanks for the update. The API additions (and clarification) look good to me. Do you have other reviewers for the impl change and tests?

Reply:
Thanks again, Alan. I'll ask around for help.